sync/statesync/code_syncer.go

```diff
@@ -47,96 +49,67 @@ type CodeSyncerConfig struct {
 // codeSyncer syncs code bytes from the network in a seprate thread.
 // Tracks outstanding requests in the DB, so that it will still fulfill them if interrupted.
 type codeSyncer struct {
-	lock sync.Mutex
-
-	CodeSyncerConfig
+	db                       ethdb.Database
+	client                   statesyncclient.Client
+	maxOutstandingCodeHashes int
+	numCodeFetchingWorkers   int
 
+	lock                  sync.Mutex
 	outstandingCodeHashes set.Set[ids.ID]  // Set of code hashes that we need to fetch from the network.
+	dbCodeHashes          []common.Hash    // Channel of code hashes stored in the database.
 	codeHashes            chan common.Hash // Channel of incoming code hash requests
-
-	// Used to set terminal error or pass nil to [errChan] if successful.
-	errOnce sync.Once
-	errChan chan error
-
-	// Passed in details from the context used to start the codeSyncer
-	cancel context.CancelFunc
-	done   <-chan struct{}
+	open                  chan struct{}    // Signal that the code syncer is open and ready to accept requests.
+	done                  <-chan struct{}
 }
 
 // newCodeSyncer returns a code syncer that will sync code bytes from the network in a separate thread.
-func newCodeSyncer(config CodeSyncerConfig) *codeSyncer {
-	return &codeSyncer{
-		CodeSyncerConfig:      config,
-		codeHashes:            make(chan common.Hash, config.MaxOutstandingCodeHashes),
-		outstandingCodeHashes: set.NewSet[ids.ID](0),
-		errChan:               make(chan error, 1),
+func newCodeSyncer(db ethdb.Database, client statesyncclient.Client, maxOutstandingCodeHashes, numCodeFetchingWorkers int) (*codeSyncer, error) {
+	syncer := &codeSyncer{
+		db:                       db,
+		client:                   client,
+		maxOutstandingCodeHashes: maxOutstandingCodeHashes,
+		numCodeFetchingWorkers:   numCodeFetchingWorkers,
+		codeHashes:               make(chan common.Hash, maxOutstandingCodeHashes),
+		outstandingCodeHashes:    set.NewSet[ids.ID](0),
```

Comment:
Also not related to the PR, but there is no need to initialize this set (with a zero length). The zero value of a set is valid to use.
```suggestion
```